src/Symfony/Component/AssetMapper/MappedAsset.php

```diff
@@ -20,119 +20,72 @@
  */
 final class MappedAsset
 {
-    private string $publicPath;
-    private string $publicPathWithoutDigest;
+    public readonly string $sourcePath;
+    public readonly string $publicPath;
+    public readonly string $publicPathWithoutDigest;
+    public readonly string $publicExtension;
+    public readonly string $content;
+    public readonly string $digest;
+    public readonly bool $isPredigested;
+
     /**
-     * @var string the filesystem path to the source file
+     * @var AssetDependency[]
      */
-    private string $sourcePath;
-    private string $content;
-    private string $digest;
-    private bool $isPredigested;
-    /** @var AssetDependency[] */
     private array $dependencies = [];
-    /** @var string[] */
-    private array $fileDependencies = [];
-
-    public function __construct(private readonly string $logicalPath)
-    {
-    }
-
-    public function getLogicalPath(): string
```

Comment:
replacing most getters+setters by readonly props (except for "dependencies" arrays, which need mutability)